### PR TITLE
Added support for AddData event

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/api/gql/fragments/fragment-metadata-block.graphql [hunk 1]
```diff
@@ -51,3 +42,24 @@ fragment MetadataBlock on MetadataBlockExtended {
     }
   }
 }
+
+fragment CustomSetWatermark on SetWatermark {
```
Comment (zaychenko-sergei) [inline src/app/api/gql/fragments/fragment-metadata-block.graphql:46]: Kill your previous fragment drafts

File: src/app/api/kamu.graphql.interface.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // THIS FILE IS GENERATED, DO NOT EDIT!
-import { gql } from "@apollo/client/core";
+import { gql } from "apollo-angular";
```
Comment (zaychenko-sergei) [inline src/app/api/kamu.graphql.interface.ts:2]: This line should not change

File: src/app/dataset-block/metadata-block/components/event-details/components/add-data-event/add-data-event.component.html [hunk 3]
```diff
@@ -0,0 +1,144 @@
+<div class="d-flex flex-column gap-4">
+    <div
+        *ngIf="event?.inputCheckpoint"
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/components/add-data-event/add-data-event.component.html:3]: `event` cannot be null, looking on it's type. Why do you need those ? checks

File: src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts [hunk 4]
```diff
@@ -31,6 +36,14 @@ export class EventDetailsComponent extends BaseComponent implements OnInit {
         );
     }
 
+    public get isAddDataEvent(): boolean {
+        return this.block.event.__typename === SupportedEvents.AddData;
+    }
+
+    public get addDataEvent(): AddData {
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts:43]: Should be `AddDataEventFragment`

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
Please work on the tooltips using JSON schema description from https://github.com/open-data-fabric/open-data-fabric/tree/master/schemas

File: src/app/api/kamu.graphql.interface.ts (hunk 2 above)
Comment (dmitriy-borzenko) [reply src/app/api/kamu.graphql.interface.ts:2]: fixed

File: src/app/api/gql/fragments/fragment-metadata-block.graphql (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/api/gql/fragments/fragment-metadata-block.graphql:46]: Killed

File: src/app/dataset-block/metadata-block/components/event-details/components/add-data-event/add-data-event.component.html (hunk 3 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-block/metadata-block/components/event-details/components/add-data-event/add-data-event.component.html:3]: Removed ckecks "?"

File: src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts [hunk 5]
```diff
@@ -31,6 +36,14 @@ export class EventDetailsComponent extends BaseComponent implements OnInit {
         );
     }
 
+    public get isAddDataEvent(): boolean {
```
Comment (zaychenko-sergei) [inline src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts:39]: Please improve unit test coverage for `EventDetailsComponent` in general

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**